### ISSUE TITLE
Automated checks does not detect contrast issues with icons

### ISSUE THREAD

**Issue by cabillin** [microsoft/accessibility-insights-web#6973]
## Describe the bug

Automated checks does not detect contrast issues with icons

**To Reproduce**

1. Open CodePen linked below
2. Notice that the icon & text have extremely poor contrast against the background
3. Open A11yInsights plugin
4. Click "Ad Hoc Tools"
5. Toggle on "Automated checks"
6. Grant A11yInsights plugin access to iframe
7. Notice that only the text's color contrast is flagged as an issue (not the icon's color contrast)

## CodePen repro example

https://codepen.io/KnetTravlr/pen/abPVayo

## Expected behavior

Color contrast issues with icon's colors should also be flagged by the "Automated checks"

## Screenshots

![image](https://github.com/microsoft/accessibility-insights-web/assets/71525014/4a557501-6f0e-4ada-8e0c-c465211ccde0)

## Context (please complete the following information)

- OS Name & Version: Windows 11 Enterprise Insider Preview 23550.1000
- AI-Web Version & Environment: ???
- Browser Version: Microsoft Edge version 117.0.0.0
- Target Page: Microsoft internal page

## Are you willing to submit a PR?

No

## Did you search for similar existing issues?

Yes

## Additional context

Accessibility Insights for Web 2.39.0 (axe-core 4.7.2)

**Comment by DaveTryon** [comment]
Hi, @cabillin! Icon color contrast issues are reported on the "Needs Review" page, which is where we report results that require human evaluation besides what the automated tool can provide. I'll dig a little deeper into why icons are handled differently, then get back to you.

**Comment by DaveTryon** [comment]
@cabillin, I did some more research on this. Accessibility Insights for web uses axe-core as its central engine, and axe-core uses the same color contrast rule for text and icons. The challenge is that icons may be a required part of the UI or they may be purely decorative, and there is no standard to identify which scenario applies to a given icon. We made the decision to include icon color contrast errors in the Needs Review category because a human needs to investigate it and make the final determination.

I'll close out this issue since we don't plan on make any changes from the current behavior.

Thanks for using Accessibility Insights!
